### DISCUSSION TITLE
VS CODE NOT GIVING ANY OUTPUT FOR C++ CODES

### DISCUSSION THREAD

**Discussion by BLANK0104** [microsoft/vscode-cpptools#10621] [Q&A]
So as the title says, my VS code is not giving me output for even a code like cout<<"Helloworld". I watched a bunch of videos and tried to solve the issue but it was without any success. The code keeps running and is opened in an external terminal and the terminal closes as soon as it is opened and if I try to run a code that requires an input then the terminal closes as soon as it gets the input, so it would be great if anyone can help me with this issue .

**Comment by bobbrow** [comment]
Have you tried running in the integrated terminal?  Set the `"console"` setting in your `launch.json` to `"integratedTerminal"`

**Comment by BLANK0104** [comment]
Hi bobbrow,
sorry for the late reply. I was busy with my exams so I couldn't test it out.
Now according to your solution, I can't find the term console in my launch.json. I can only find the term "externalconsole" which expects a true or false value.
Please have a look at my launch.json file:

{
    "version": "0.2.0",
    "configurations": [
      {
        "name": "g++.exe build and debug active file",
        "type": "cppdbg",
        "request": "launch",
        "program": "${fileDirname}\\${fileBasenameNoExtension}.exe",
        "args": [],
        "stopAtEntry": false,
        "cwd": "${workspaceFolder}",
        "environment": [],
        "externalConsole": true, //set to true to see output in cmd instead
        "MIMode": "gdb",
        "miDebuggerPath": "C:\\MinGW\\bin\\gdb.exe",
        "setupCommands": [
          {
            "description": "Enable pretty-printing for gdb",
            "text": "-enable-pretty-printing",
            "ignoreFailures": true
          }
        ],
        "preLaunchTask": "g++.exe build active file"
      },
      {
        "name": "g++ build & run active file",
        "type": "cppdbg",
        "request": "launch",
        "program": "${fileDirname}\\${fileBasenameNoExtension}.exe",
        "args": [],
        "stopAtEntry": false,
        "cwd": "${workspaceFolder}",
        "environment": [],
        "externalConsole": false, //set to true to see output in cmd instead
        "MIMode": "gdb",
        "miDebuggerPath": "C:\\MinGW\\bin\\gdb.exe",
        "setupCommands": [
          {
            "description": "Enable pretty-printing for gdb",
            "text": "-enable-pretty-printing",
            "ignoreFailures": true
          }
        ],
        "preLaunchTask": "g++ build & run active file"
      }
      ]
  }

**Reply by sean-mcmanus** [reply]
Yeah, can you set "externalConsole" to false? The "console" setting appears to be for "cppvsdbg" debug types.

**Reply by BLANK0104** [reply]
if i set it to false then i stop receing any sort of output at all.
with externalconsole set as true, i can receive output with system("pause"); but with externalconsole set as false, i dont get any output at all, not even in the integrated terminal  
![image](https://user-images.githubusercontent.com/115344345/223330106-59f91671-7655-4028-84a7-acbfa32da6d0.png)

**Reply by sean-mcmanus** [reply]
You need to flush your output stream via adding a `<< endl;` to the `cout` expression or `\n` (to the string literal).

**Reply by BLANK0104** [reply]
it doesnt work. endl and \n are used to create new line and end line. even with using those in code, i get no output in terminal.

**Reply by sean-mcmanus** [reply]
The endl and \n also cause the output stream to be flushed, as would `<< std::flush`. If that doesn't fix the issue, then I'm not sure. @WardenGnaw Any other ideas?

**Reply by BLANK0104** [reply]
![image](https://user-images.githubusercontent.com/115344345/223943899-2d6603da-d70f-41ef-a768-57caa9a515d1.png)

no output.

**Reply by sean-mcmanus** [reply]
Yeah, I don't know, it works for me -- maybe you can try updating to a newer mingw version?

**Reply by BLANK0104** [reply]
how?
and i believe that i am already on the latest version.

**Reply by sean-mcmanus** [reply]
`C:\MinGW\bin\gcc.exe --version` should output the version. Version 10+ is recommended and 4 or less is not recommended. You can get newer versions at https://www.msys2.org/ .

**Reply by BLANK0104** [reply]
i did install the latest version from mysys. I was on version 6.3.0 and from mysys i got version 12.2.0-10 but for some reasons, when i put gcc --version, it still shows me version 6.3.0 and in mysys, it says command not found

![image_2023-03-18_221335206](https://user-images.githubusercontent.com/115344345/226120815-00413cee-a1b8-45cb-b952-489cb8cbecdf.png)

**Reply by sean-mcmanus** [reply]
You may need to add the path to the mingw binary's folder to your PATH, such that it's found before the older one.
